Implement bracket creation functionality

NikolasGeorgaklis opened the pull request:
User can now select teams to win both first round and future theoretical matchups.

- Matchups displayed in tableview cells in CreateBracketViewController, each cell shows a matchup
- Cells are separated by region into sections until final 4
- Each cell has a segmented controller with two options, one for each team, user can select who they predict will win a given match.
- Navbar contains segmented controller which indicates the round of the tournament.
- All selections are put into Parse.

NikolasGeorgaklis committed:
Implement bracket creation functionality

User can now select teams to win both first round and future theoretical matchups.

- Matchups displayed in tableview cells in CreateBracketViewController, each cell shows a matchup
- Cells are separated by region into sections until final 4
- Each cell has a segmented controller with two options, one for each team, user can select who they predict will win a given match.
- Navbar contains segmented controller which indicates the round of the tournament.
- All selections are put into Parse.

vaguilar commented on NikoBracket/CreateBracketViewController.m:
```diff
@@ -61,87 +80,296 @@ - (void)fetchData{
 - (NSInteger)tableView:(nonnull UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+   
+    switch (self.roundControl.selectedSegmentIndex) {
+        case 0:
+            return 8;
+            break;
+        case 1:
+            return 4;
+            break;
+        case 2:
+            return 2;
+            break;
+        case 3:
+            return 1;
+            break;
+        case 4:
+            return 2;
+            break;
+        default:
```
It would be helpful to have the round numbers as constants at the top of the file (similar to the API endpoint) and replace the usages of them here and throughout this file with the constants. Something like:

```
static int kRound1 = 0;
...
static int kSemiFinal = 4;
static int kFinal = 5;
```

vaguilar commented on NikoBracket/CreateBracketViewController.m:
```diff
@@ -175,24 +411,136 @@ - (UIView *)tableView:(UITableView *)tableView viewForHeaderInSection:(NSInteger
+        }
+    }
+    
+    matchups[1] = round2;
+    matchups[2] = round3;
+    matchups[3] = round4;
```
I think this may just be accidentally working. We should check the size of the `matchups` array, and if its not big enough then handle this case

vaguilar commented on NikoBracket/CreateBracketViewController.m:
```diff
@@ -175,24 +411,136 @@ - (UIView *)tableView:(UITableView *)tableView viewForHeaderInSection:(NSInteger
+        NSArray *game = round1[i];
+        NSArray *game2 = round1[i+1];
 
+        NSString *winner1 = game[[game[2] intValue]];
+        NSString *winner2 = game2[[game2[2] intValue]];
```
it would also be helpful to make clear what each `game` looks like, by adding a comment and explaining what each value in the array means somewhere in this file

vaguilar commented:
looks good, left a few comments.

NikolasGeorgaklis committed:
User can now select teams to win both first round and future theoretical matchups.(2)

 - Matchups displayed in tableview cells in CreateBracketViewController, each cell shows a matchup
Cells are separated by region into sections until final 4
 - Each cell has a segmented controller with two options, one for each team, user can select who they predict will win a given match.
 - Navbar contains segmented controller which indicates the round of the tournament.
 - All selections are put into Parse.
 - I declared the rounds at the top of the file as constants
 - I changed the way the rounds are added to the matchups arrays
 - Added a comment explaining "game" objects.

vaguilar commented on NikoBracket/CreateBracketViewController.m:
```diff
@@ -427,6 +433,7 @@ - (NSMutableArray *) initializeUserBracket:(NSMutableArray *) matchups {
     NSMutableArray *round4 = [[NSMutableArray alloc] init];
     
     for (int i = 0; i < round1.count - 1; i+=2) {
+        //each "game" object contains 3 elements: two 3-5 letter abbreviations, one for the home team, one for the away team, and the third element is a 0 or 1 indicating the user’s choice for who they believe will win the game
```
👍

vaguilar commented:
looks good to me!

NikolasGeorgaklis committed:
Merge branch 'main' into feature_CreateBracket
